crypto/cmp/cmp_client.c

```diff
@@ -171,20 +172,27 @@ static int send_receive_check(OSSL_CMP_CTX *ctx, const OSSL_CMP_MSG *req,
 #endif
 
     OSSL_CMP_log1(INFO, ctx, "sending %s", req_type_string);
-    err = (*transfer_cb)(ctx, req, rep);
+    *rep = (*transfer_cb)(ctx, req);
     ctx->msgtimeout = msgtimeout; /* restore original value */
 
-    if (err != 0) {
-        if (err == CMP_R_FAILED_TO_RECEIVE_PKIMESSAGE
-                || err == CMP_R_READ_TIMEOUT
-                || err == CMP_R_ERROR_DECODING_MESSAGE) {
+    if (*rep == NULL) {
+        int err = ERR_peek_error();
+        int lib = ERR_GET_LIB(err);
+        int reason = ERR_GET_REASON(err);
+        if ((lib == ERR_LIB_BIO && (reason == BIO_R_TRANSFER_ERROR
```

Comment:
switch(reason) ?

Reply:
`switch(reason)` would be a bit hard to apply here as I mostly check for _non-_equality.

Reply:
Do you have a concrete suggestion to make this code better readable?